linkerd/stack/src/make_ready.rs

```diff
@@ -1,18 +1,30 @@
-use futures::{try_ready, Future, Poll};
+use futures::{ready, TryFuture};
 use linkerd2_error::Error;
+use pin_project::{pin_project, project};
+use std::future::Future;
 use std::marker::PhantomData;
-use tower::util::Ready;
+use std::pin::Pin;
+use std::task::{Context, Poll};
 
 #[derive(Debug)]
 pub struct MakeReadyLayer<Req>(PhantomData<fn(Req)>);
 
 #[derive(Debug)]
 pub struct MakeReady<M, Req>(M, PhantomData<fn(Req)>);
 
+#[pin_project]
 #[derive(Debug)]
-pub enum MakeReadyFuture<F, S, Req> {
-    Making(F),
-    Ready(Ready<S, Req>),
+pub struct MakeReadyFuture<F, S, Req> {
```

Comment:
Similar question here about making this a struct with an enum field, instead of keeping it as an enum.